gef.py

```diff
@@ -11133,27 +11156,46 @@ def __setup_qemu(self) -> bool:
         return True
 
     def __setup_remote(self) -> bool:
+        success = True
+
         # get the file
-        fpath = f"/proc/{self.pid}/exe"
-        if not self.sync(fpath, str(self.file)):
-            err(f"'{fpath}' could not be fetched on the remote system.")
-            return False
+        if not self.sync(str(self.file)):
+            warn(f"'{str(self.file)}' could not be fetched on the remote system.")
+            success = False
 
-        # pseudo procfs
-        for _file in ("maps", "environ", "cmdline"):
-            fpath = f"/proc/{self.pid}/{_file}"
-            if not self.sync(fpath):
-                err(f"'{fpath}' could not be fetched on the remote system.")
-                return False
+        fpath = f"/proc/{self.pid}/maps"
+        if not self.sync(fpath):
+            warn(f"'{fpath}' could not be fetched on the remote system.")
+            success = False
 
-        # makeup a fake mem mapping in case we failed to retrieve it
-        maps = self.root / f"proc/{self.pid}/maps"
-        if not maps.exists():
+            # makeup a fake mem mapping in case we failed to retrieve it
+            maps = self.root / fpath
             with maps.open("w") as fd:
                 fname = self.file.absolute()
                 mem_range = "00000000-ffffffff" if is_32bit() else "0000000000000000-ffffffffffffffff"
                 fd.write(f"{mem_range} rwxp 00000000 00:00 0                    {fname}\n")
-        return True
+
+        fpath = f"/proc/{self.pid}/environ"
+        if not self.sync(fpath):
+            warn(f"'{fpath}' could not be fetched on the remote system.")
+            success = False
+
+            # makeup a fake mem mapping in case we failed to retrieve it
+            environ = self.root / fpath
+            with environ.open("wb") as fd:
+                fd.write(b"PATH=/bin\x00HOME=/tmp\x00")
+
+        fpath = f"/proc/{self.pid}/cmdline"
```

Comment:
IT would be nice if we could just tell gef to use a buffer instead of having to write stuff to disk when faking stuff. It would be even nicer if we didn't have to fake it.